app/models/computer_vision_demo_upload.rb

```diff
@@ -33,22 +33,8 @@ def extract_metadata( path )
     return unless path
     self.metadata ||= { }
     begin
-      if photo_content_type =~ /jpe?g/i && exif = EXIFR::JPEG.new( path )
-        self.metadata.merge!( exif.to_hash )
-        xmp = XMP.parse( exif )
-        if xmp && xmp.respond_to?( :dc ) && !xmp.dc.nil?
-          self.metadata[:dc] = { }
-          xmp.dc.attributes.each do |dcattr|
-            begin
-              unless xmp.dc.send(dcattr).blank?
-                self.metadata[:dc][dcattr.to_sym] = xmp.dc.send( dcattr )
-              end
-            rescue ArgumentError
-              # XMP does this for some DC attributes, not sure why
-            end
-          end
-        end
-      end
+      exif_data = ExifMetadata.new(path: path, type: photo_content_type ).extract
```

Comment:
FYI, our general Ruby style is to include a space within parens, so this should be `ExifMetadata.new( path: path, type: photo_content_type )`